src/lib/security/selinux.rb

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Security
+  # Class for handling SELinux kernel params
+  class Selinux
+    Yast.import "Bootloader"
+
+    MODULE_NAME = "selinux".freeze
+    private_constant :MODULE_NAME
+
+    ENABLE_VALUE = "1".freeze
+    private_constant :ENABLE_VALUE
+
+    def enabled?
+      param(:security) == MODULE_NAME && param(:selinux) == ENABLE_VALUE
+    end
+
+    def enforcing?
+      return false unless enabled?
+
+      param("enforcing") == ENABLE_VALUE
+    end
+
+    def permissive?
+      return false unless enabled?
+
+      !enforcing?
+    end
+
+    def disabled?
+      !enabled?
+    end
```

Comment:
what I am missing there is method like `policy` or `value` which can return any value from  list `[:permissive, :enforcing, :disabled]` So it will be easier to get current value for widget. Otherwise I have to ask for each one.

Also what would be useful is to have human name mapping to allow display of value like https://github.com/yast/yast-installation/pull/906/files#diff-21c7a291eed8a2f9df01ced4faea7e8be674002dc095aa746f12f75c018014fcR144

Reply:
and when we have that method value or policy, it would be nice to have also writter, so value from combobox can be directly just store in widget and do not need to check value and call respective method.